### PR TITLE
Bug 1698152: fix image source extraction with trailing "/."

### PR DESCRIPTION
**Pull request by adambkaplan**
* Fix copy behavior if the image source path ends with "/."
* Add unit tests for copying regular files and symlinks

/assign @coreydaley 

/cc @bparees 

fyi @openshift/sig-developer-experience

### CODE REVIEW

**Comment by bparees** [comment]
@adambkaplan whew, i was worried this might be deeper in buildah's image mounting/copying logic.

**Comment by bparees** [comment]
(also i can't wait for someone to complain that we are not preserving their symlinks...  i'm pretty sure we've gone back and forth on this behavior in a variety of our file copying/moving codepaths, multiple times)

**Comment by adambkaplan** [comment]
/hold

Trying to make sure this actually works via e2e tests

**Comment by adambkaplan** [comment]
@bparees your fears may have been justified. See https://github.com/openshift/origin/pull/22456#issuecomment-479285125

File: pkg/build/builder/source_test.go [hunk 1]
```diff
@@ -304,3 +304,72 @@ func TestCloneFromBranch(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyImageSourceFromFilesystem(t *testing.T) {
```
Comment (coreydaley) [inline pkg/build/builder/source_test.go]: This test should create the symlink in the src directory to a file in the link directory, then copy the src directory to the destination directory.  That would seem to follow more along the lines of what we would actually do?  Might just be me, but it seemed confusing the way it is done now, trying to figure out why things were being created/symlinked to other places that didn't make sense.

Comment (adambkaplan) [reply pkg/build/builder/source_test.go]: @coreydaley test has been updated to be table-driven. Hopefully it's clearer.

**Pull request merged**